android/src/main/java/com/twilio/twilio_voice/AnswerJavaActivity.java

```diff
@@ -231,7 +234,7 @@ private void endCall() {
             intent.setAction(Constants.ACTION_CANCEL_CALL);
 
             this.startActivity(intent);
-            finish();
+            finishAndRemoveTask();
```

Comment:
action_cancel_call is when the caller hangs up during an incoming call. 

If the receiver hangs up we have the event action_call_ended and perform finishAndRemoveTask().